src/topic_training/src/Begin_Here.md

```diff
@@ -0,0 +1,122 @@
+# WRoverPlayground - Topic Training
+Now that you have completed the Tools Training, you are ready to learn about ROS topics. From the Setup Training, you should be familiar with:
+- the basics of ROS Nodes
+- rosnode
+- and roscore.
+
+Once you complete this training, you should be familiar with:
+- the purpose of topics
+- the relationship between Publishers and Subscribers
+- setting up Publishers and Subscribers
+- running Publishers and Subscribers
+- rqt_graph
+- rqt_plot
+- custom topic types
+- and rostopic type {topic}.
+
+## The Purpose of Topics
+The word "topic" sounds very general, but it is actually a specific concept that allows ROS nodes to communicate with each other. This is extremely important because one node cannot possibly contain all the information a rover would need.
+
+Topics allow us to specialize nodes, so they can execute their own code/tasks, while also sending and receiving information. For example, a node could receive data and send it to another one, a node could receive and analyze data and have another node do some action with it, a node could receive data from multiple nodes, etc.
+
+## Publisher/Subscriber Relationship
+The Publisher and Subscriber are both nodes that are connected by a topic, allowing them to send messages between themselves. The Publisher node publishes messages on the topic, while the Subscriber node subscribes to the same topic to receive the Publisher's messages. 
+
+Publishers and Subscribers do not need to be in separate files. In fact, if the Publisher and Subscriber rely on each other's data, it may be better to keep them in the same file. An example of the Publisher and Subscriber residing in the same file will be provided in the next section.
+
+## Setting up a Publisher and Subscriber
+Within this directory, you will find two files called Demo(.cpp or .py). These files will show a basic class containing both a Publisher and Subscriber. There will be comments throughout the files that explain the purpose of specific sections of the code.
+
+Note that there are quite a few sections of code that are marked as "Optional." You do not need to include everything in the examples when creating your own code. They are simply showing one way of setting up a Publisher and Subscriber in the same file.
+
+One file will be written in C++, and the other will be in Python. I recommend discussing with one of the Software Leads to determine which part of the rover you would like to be assigned to. This will help you decide which language to focus on for this training. However, being familiar with both languages will not hurt.
+
+Go through one of the files, making sure you understand all the code and comments. Come back to this file once you feel ready to move on.
+
+## Running a Publisher and Subscriber
+Before you are able to run the example code, find the CMakeLists.txt file that is outside of the src directory but still inside of the topic_training directory. 
```

Comment:
Yes, I would avoid this.  We are probably going to have the custom types be pre-built and just show them how it's done/direct them to relevant code to show them how it would be done.

Reply:
Fixed in ead3ab7.

Reply:
Fixed in ead3ab7.